### PR TITLE
Install logging and ES operator and pre-configure curator

### PR DESCRIPTION
**Pull request by jewzaam**
In v4 we'll support the customer creating a ClusterLogging CR to opt-in to the logging stack running on their cluster.  It will run on the workers.  Ability to do this requires update to dedicated-admin permissions which will be coming soon.

### CODE REVIEW

**Comment by fahlmant** [comment]
What happens if the namespace creation fails?

File: hack/00-osd-managed-cluster-config.selectorsyncset.yaml.tmpl [hunk 1]
```diff
@@ -706,6 +706,151 @@ objects:
...
+      metadata:
+        labels:
+          csc-owner-name: installed-redhat-openshift-logging
+          csc-owner-namespace: openshift-marketplace
+        name: cluster-logging
+      spec:
+        channel: preview
+        installPlanApproval: Automatic
+        name: cluster-logging
+        source: installed-redhat-openshift-logging
```
Comment (rogbas) [inline hack/00-osd-managed-cluster-config.selectorsyncset.yaml.tmpl]: I don't see a catalogsource in this repo with that name, is it supposed to be pre-installed in the clusters?

Comment (jewzaam) [reply hack/00-osd-managed-cluster-config.selectorsyncset.yaml.tmpl]: Good catch, looking into this.  May be installed as part of installation via the console UI.  I will update this PR when I have an update.

Comment (jewzaam) [reply hack/00-osd-managed-cluster-config.selectorsyncset.yaml.tmpl]: Fixed.  OperatorHub in console drops a CatalogSourceConfig, which I have included now for the 2 operators.  Also had missed the OperatorGroup and had a broken Subscription for ES.

**Comment by jeremyeder** [comment]
And the example CR in the [product docs](https://docs.openshift.com/container-platform/4.1/logging/efk-logging-deploying.html) is what we will point users to?

Any tweaks to the storage aspects necessary?

```
apiVersion: "logging.openshift.io/v1"
kind: "ClusterLogging"
metadata:
  name: "instance" 
  namespace: "openshift-logging"
spec:
  managementState: "Managed"  
  logStore:
    type: "elasticsearch"  
    elasticsearch:
      nodeCount: 3 
      storage:
        storageClassName: gp2
        size: 200G
      redundancyPolicy: "SingleRedundancy"
  visualization:
    type: "kibana"  
    kibana:
      replicas: 1
  curation:
    type: "curator"  
    curator:
      schedule: "30 3 * * *"
  collection:
    logs:
      type: "fluentd"  
      fluentd: {}
```

**Comment by dak1n1** [comment]
Just wanted to remind you to run `make` if you haven't already. I do see one generated template file, so maybe that means you already ran it.

**Pull request merged**